.github/workflows/publish.yml

```diff
@@ -3,31 +3,30 @@ name: Publish
 on:
   push:
     tags:
-      - '*'
+      - 'v*'
+  # Manual run
+  workflow_dispatch:
+    inputs:
+      tag:
+        required: true
+        type: string
+        description: 'Tag to release'
 
 jobs:
-  publish:
-    name: Publish to npm
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+  release:
+    name: release
     runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
 
-      - uses: volta-cli/action@v4
+    steps:
+      - uses: actions/checkout@v4
         with:
-          node-version: 16.x
-          registry-url: 'https://registry.npmjs.org'
-
-      - name: Install Dependencies
-        run: yarn install --frozen-lockfile
+          ref: ${{ github.event.inputs.tag || github.ref_name }}
 
-      - name: auto-dist-tag
-        run: npx auto-dist-tag --write
+      - uses: wyvox/action-setup-pnpm@v3
```

Comment:
Does this set registry-url? Otherwise publishing will not work

Reply:
not by default, no. Just added.

Should specifying a registry-url be default? any downsides to that?

Reply:
I don't typically do it by default, it changes whether setup-node stores credentials or not